src/controllers/problem.ts

```diff
@@ -101,7 +101,7 @@ const problemController = {
                     };
 
 
-                    if (req.body.source && req.body.problemMetadata && req.body.problemMetadata.platformProblemId) {
+                    if (req.body.source && req.body.problemMetadata) {
```

Comment:
can be removed since they are required in the validator

Reply:
only source is required in validator. See my comments